pkg/arvo/app/dojo.hoon

```diff
@@ -679,11 +704,16 @@
       [%face ^]  a(q $(a q.a))
       [%cell ^]  a(p $(a p.a), q $(a q.a))
       [%fork *]  a(p (silt (turn ~(tap in p.a) |=(b=type ^$(a b)))))
-      [%hint *]  ?.  ?=(%know -.q.p.a)  $(a q.a)
-                 ?@  p.q.p.a  [(cat 3 '#' mark.p.q.p.a)]~
-                 [(rap 3 '#' auth.p.q.p.a (spat type.p.q.p.a) ~)]~
+      [%hint *]  ?+    q.p.a  $(a q.a)
+                     [%know *]
+                   ?@  p.q.p.a  [(cat 3 '#' mark.p.q.p.a)]~
+                   [(rap 3 '#' auth.p.q.p.a (spat type.p.q.p.a) ~)]~
+                 ::
+                     [%help *]
+                   [summary.crib.p.q.p.a]~
+                 ==
       [%core ^]  `wain`/core
-      [%hold *]  a(p $(a p.a))
+      [%hold *]  $(a (~(play ut p.a) q.a))
```

Comment:
Playing `%hold`s (i.e. what we're doing here, not sure if there's an accepted term for this) can take a long time and often this just crashes. Is there a better option? I suppose it could just give up after some chosen depth, but this isn't really user friendly. It ought to "just work". Besides limiting by depth, the only other idea I have is Clay building a cache of doccords.

Reply:
Since addressing this sort of issue is one of the main goals of the doccords apprenticeship, my opinion now is that its fine to ship this since its an experimental feature seeking a good interface, and its fine to have pain points in the first attempt at an interface.